include/boost/geometry/extensions/gis/io/wkb/detail/writer.hpp

```diff
@@ -0,0 +1,217 @@
+				boost::ignore_unused_variable_warning(iter);
+				boost::ignore_unused_variable_warning(byte_order);
+			}
+		};
+
+		template <typename P>
+		struct point_writer
+		{
+			template <typename OutputIterator>
+			static bool write(const P& point, OutputIterator& iter,
+				byte_order_type::enum_t byte_order)
+			{
+				// write endian type
+
+				value_writer<uint8_t>::write(byte_order, iter, byte_order);
+
+				// write geometry type
+				uint32_t type = geometry_type<P>::get();
+				value_writer<uint32_t>::write(type, iter, byte_order);
+
+				// write point's x, y, z
+
+				writer_assigner<P, 0, dimension<P>::value>::run(point, iter, byte_order);
+
+				return true;
+			}
+		};
+
+		template <typename L>
+		struct linestring_writer
+		{
+			template <typename OutputIterator>
+			static bool write(const L& linestring, OutputIterator& iter,
+				byte_order_type::enum_t byte_order)
+			{
+				// write endian type
+
+				value_writer<uint8_t>::write(byte_order, iter, byte_order);
+
+				// write geometry type
+				uint32_t type = geometry_type<L>::get();
+				value_writer<uint32_t>::write(type, iter, byte_order);
+
+				// write num points
+
+				uint32_t num_points = boost::size(linestring);
+				value_writer<uint32_t>::write(num_points, iter, byte_order);
+
+				typedef typename point_type<L>::type point_type;
+
+				for(boost::range_iterator<const L>::type point_iter = boost::begin(linestring); point_iter != boost::end(linestring); ++point_iter)
+				{
+					// write point's x, y, z
+					writer_assigner<point_type, 0, dimension<point_type>::value>::run(*point_iter, iter, byte_order);
+				}
+
+				return true;
+			}
+		};
+
+		template <typename P>
+		struct polygon_writer
+		{
+			template <typename OutputIterator>
+			static bool write(const P& polygon, OutputIterator& iter,
+				byte_order_type::enum_t byte_order)
+			{
+				// write endian type
+
+				value_writer<uint8_t>::write(byte_order, iter, byte_order);
+
+				// write geometry type
+				uint32_t type = geometry_type<P>::get();
+				value_writer<uint32_t>::write(type, iter, byte_order);
+
+				// write num rings
+
+				uint32_t num_rings = 1 + boost::geometry::num_interior_rings(polygon);
+				value_writer<uint32_t>::write(num_rings, iter, byte_order);
+
+				typedef typename point_type<P>::type point_type;
+
+				// write exterior ring
+				typedef typename boost::geometry::ring_type<const P>::type ring_type;
+				ring_type exterior_ring = boost::geometry::exterior_ring(polygon);
+
+				uint32_t num_exterior_ring_points = boost::geometry::num_points(exterior_ring);
+				value_writer<uint32_t>::write(num_exterior_ring_points, iter, byte_order);
+
+				for(
+					boost::range_iterator<const ring_type>::type point_iter = boost::begin(exterior_ring); 
+					point_iter != boost::end(exterior_ring);
+					++point_iter
+					)
+				{
+					// write point's x, y, z
+					writer_assigner<point_type, 0, dimension<point_type>::value>::run(*point_iter, iter, byte_order);
```

Comment:
Here similar case, for-loop (empty line and big indentation) and template parameters of `writer_assigner`.

Btw there are some guidelines for developers, here: https://github.com/boostorg/geometry/wiki/Guidelines-for-Developers